library/src/main/java/me/mattlogan/library/ViewStack.java

```diff
@@ -67,50 +91,63 @@ public void saveToBundle(Bundle bundle, String tag) {
-            checkNotNull(viewFactory, "viewFactory == null");
-            pushWithoutNotifyingListeners(viewFactory);
+        for (Integer layoutResource : savedStack) {
+            pushWithoutNotifyingListeners(layoutResource);
         }
         notifyListeners();
     }
 
     /**
-     * Pushes a View, created by the provided ViewFactory, onto the navigation stack
+     * Pushes a View id onto the navigation stack and inflates it in the container
      *
-     * @param viewFactory responsible for the creation of the next View in the navigation stack
-     * @return the provided ViewFactory (to comply with the Java Stack API)
+     * @param layoutId The layout id to inflate into the parent {@link ViewGroup}.
+     * @return the provided layout id (to comply with the Java Stack API)
      */
-    public ViewFactory push(ViewFactory viewFactory) {
-        checkNotNull(viewFactory, "viewFactory == null");
-        pushWithoutNotifyingListeners(viewFactory);
+    public int push(@LayoutRes int layoutId) {
+        pushWithoutNotifyingListeners(layoutId);
         notifyListeners();
-        return viewFactory;
+        return layoutId;
     }
 
-    private void pushWithoutNotifyingListeners(ViewFactory viewFactory) {
-        stack.push(viewFactory);
-        View view = viewFactory.createView(container.getContext(), container);
+    private void pushWithoutNotifyingListeners(@LayoutRes int layoutId) {
+        stack.push(layoutId);
+        View view = pancakesViewInflator.inflateView(layoutId, container);
         container.addView(view);
         setBelowViewVisibility(View.GONE);
     }
 
     /**
-     * Pushes a View, created by the provided ViewFactory, onto the navigation stack and animates
-     * it using the Animator created by the provided AnimatorFactory
+     * Pops the top View off the navigation stack
+     *
+     * @return The id of the layout on the top of the stack, or SINGLE_VIEW if the stack is to
+     * be finished.
+     */
+    public int pop() {
+        if (!shouldPop()) return SINGLE_VIEW;
+        int layoutId = stack.pop();
+        setBelowViewVisibility(View.VISIBLE);
+        container.removeView(peekView());
+        notifyListeners();
+        return layoutId;
+    }
+
+    /**
+     * Pushes a View, created with the provided layout id, onto the navigation stack and animates
+     * it using the Animator created by the provided {@link AnimatorFactory}
      *
-     * @param viewFactory     responsible for the creation of the next View in the navigation stack
+     * @param layoutId     The id of the view to be added to the top of the navigation stack
      * @param animatorFactory responsible for the creation of an Animator to animate the next View
      *                        onto the navigation stack
-     * @return the provided ViewFactory (to comply with the Java Stack API)
+     * @return the provided layout id (to comply with the Java Stack API)
```

Comment:
Can we actually remove that extra comment in parentheses? In hindsight it is kind of superfluous and obvious.